File: companion/internal/index/classify.go
package index

import "github.com/DoubleGremlin181/fast-travel-app/companion/internal/protocol"

// extCategory maps lowercase file extensions to FileType categories.
var extCategory = map[string]protocol.FileType{
	// image
	"jpg": protocol.FileTypeImage, "jpeg": protocol.FileTypeImage,
	"png": protocol.FileTypeImage, "gif": protocol.FileTypeImage,
	"webp": protocol.FileTypeImage, "bmp": protocol.FileTypeImage,
	"svg": protocol.FileTypeImage, "heic": protocol.FileTypeImage,
	"tiff": protocol.FileTypeImage, "ico": protocol.FileTypeImage,

	// video
	"mp4": protocol.FileTypeVideo, "mkv": protocol.FileTypeVideo,
	"mov": protocol.FileTypeVideo, "avi": protocol.FileTypeVideo,
	"webm": protocol.FileTypeVideo, "flv": protocol.FileTypeVideo,
	"wmv": protocol.FileTypeVideo, "m4v": protocol.FileTypeVideo,

	// audio
	"mp3": protocol.FileTypeAudio, "flac": protocol.FileTypeAudio,
	"wav": protocol.FileTypeAudio, "ogg": protocol.FileTypeAudio,
	"m4a": protocol.FileTypeAudio, "aac": protocol.FileTypeAudio,
	"opus": protocol.FileTypeAudio,

	// archive
	"zip": protocol.FileTypeArchive, "tar": protocol.FileTypeArchive,
	"gz": protocol.FileTypeArchive, "bz2": protocol.FileTypeArchive,
	"xz": protocol.FileTypeArchive, "7z": protocol.FileTypeArchive,
	"rar": protocol.FileTypeArchive, "zst": protocol.FileTypeArchive,

	// code
	"go": protocol.FileTypeCode, "js": protocol.FileTypeCode,
	"ts": protocol.FileTypeCode, "tsx": protocol.FileTypeCode,
	"jsx": protocol.FileTypeCode, "py": protocol.FileTypeCode,
	"rs": protocol.FileTypeCode, "java": protocol.FileTypeCode,
	"kt": protocol.FileTypeCode, "c": protocol.FileTypeCode,
	"h": protocol.FileTypeCode, "cpp": protocol.FileTypeCode,
	"hpp": protocol.FileTypeCode, "cs": protocol.FileTypeCode,
	"rb": protocol.FileTypeCode, "php": protocol.FileTypeCode,
	"sh": protocol.FileTypeCode, "json": protocol.FileTypeCode,
	"yaml": protocol.FileTypeCode, "yml": protocol.FileTypeCode,
	"toml": protocol.FileTypeCode, "xml": protocol.FileTypeCode,
	"html": protocol.FileTypeCode, "css": protocol.FileTypeCode,
	"sql": protocol.FileTypeCode,

	// document
	"pdf": protocol.FileTypeDocument, "doc": protocol.FileTypeDocument,
	"docx": protocol.FileTypeDocument, "xls": protocol.FileTypeDocument,
	"xlsx": protocol.FileTypeDocument, "ppt": protocol.FileTypeDocument,
	"pptx": protocol.FileTypeDocument, "odt": protocol.FileTypeDocument,
	"ods": protocol.FileTypeDocument, "odp": protocol.FileTypeDocument,
	"txt": protocol.FileTypeDocument, "md": protocol.FileTypeDocument,
	"rtf": protocol.FileTypeDocument, "csv": protocol.FileTypeDocument,
	"epub": protocol.FileTypeDocument,
}

// ClassifyType returns the FileType category for a given lowercase extension.
// If the extension is unknown, it returns FileTypeOther.
func ClassifyType(ext string) protocol.FileType {
	if t, ok := extCategory[ext]; ok {
		return t
	}
	return protocol.FileTypeOther
}

File: android/app/src/main/kotlin/sh/kavi/fasttravel/localsearch/index/Classify.kt
package sh.kavi.fasttravel.localsearch.index

/**
 * Maps lowercase file extensions to [FileType] categories.
 * Mirrors companion/internal/index/classify.go extCategory exactly.
 */
private val extCategory: Map<String, FileType> = mapOf(
    // image
    "jpg"  to FileType.IMAGE, "jpeg" to FileType.IMAGE,
    "png"  to FileType.IMAGE, "gif"  to FileType.IMAGE,
    "webp" to FileType.IMAGE, "bmp"  to FileType.IMAGE,
    "svg"  to FileType.IMAGE, "heic" to FileType.IMAGE,
    "tiff" to FileType.IMAGE, "ico"  to FileType.IMAGE,

    // video
    "mp4"  to FileType.VIDEO, "mkv"  to FileType.VIDEO,
    "mov"  to FileType.VIDEO, "avi"  to FileType.VIDEO,
    "webm" to FileType.VIDEO, "flv"  to FileType.VIDEO,
    "wmv"  to FileType.VIDEO, "m4v"  to FileType.VIDEO,

    // audio
    "mp3"  to FileType.AUDIO, "flac" to FileType.AUDIO,
    "wav"  to FileType.AUDIO, "ogg"  to FileType.AUDIO,
    "m4a"  to FileType.AUDIO, "aac"  to FileType.AUDIO,
    "opus" to FileType.AUDIO,

    // archive
    "zip"  to FileType.ARCHIVE, "tar" to FileType.ARCHIVE,
    "gz"   to FileType.ARCHIVE, "bz2" to FileType.ARCHIVE,
    "xz"   to FileType.ARCHIVE, "7z"  to FileType.ARCHIVE,
    "rar"  to FileType.ARCHIVE, "zst" to FileType.ARCHIVE,

    // code
    "go"   to FileType.CODE, "js"   to FileType.CODE,
    "ts"   to FileType.CODE, "tsx"  to FileType.CODE,
    "jsx"  to FileType.CODE, "py"   to FileType.CODE,
    "rs"   to FileType.CODE, "java" to FileType.CODE,
    "kt"   to FileType.CODE, "c"    to FileType.CODE,
    "h"    to FileType.CODE, "cpp"  to FileType.CODE,
    "hpp"  to FileType.CODE, "cs"   to FileType.CODE,
    "rb"   to FileType.CODE, "php"  to FileType.CODE,
    "sh"   to FileType.CODE, "json" to FileType.CODE,
    "yaml" to FileType.CODE, "yml"  to FileType.CODE,
    "toml" to FileType.CODE, "xml"  to FileType.CODE,
    "html" to FileType.CODE, "css"  to FileType.CODE,
    "sql"  to FileType.CODE,

    // document
    "pdf"  to FileType.DOCUMENT, "doc"  to FileType.DOCUMENT,
    "docx" to FileType.DOCUMENT, "xls"  to FileType.DOCUMENT,
    "xlsx" to FileType.DOCUMENT, "ppt"  to FileType.DOCUMENT,
    "pptx" to FileType.DOCUMENT, "odt"  to FileType.DOCUMENT,
    "ods"  to FileType.DOCUMENT, "odp"  to FileType.DOCUMENT,
    "txt"  to FileType.DOCUMENT, "md"   to FileType.DOCUMENT,
    "rtf"  to FileType.DOCUMENT, "csv"  to FileType.DOCUMENT,
    "epub" to FileType.DOCUMENT,
)

/**
 * Returns the [FileType] category for a given lowercase extension.
 * If the extension is unknown, returns [FileType.OTHER].
 * Mirrors companion/internal/index/classify.go ClassifyType.
 */
fun classifyType(ext: String): FileType = extCategory[ext] ?: FileType.OTHER
